### PR TITLE
Madalinap/tema7

### PR DESCRIPTION
**Pull request by pmadalina88**

### CODE REVIEW

**Review by curtnazarEliz — APPROVED** [review]

**Review by tincageorgianaalina — APPROVED** [review]

File: src/test/resources/madalinapopescu/groups.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+        <groups>
+            <run>
+                <exclude name="login"  />
+                <include name="redirect"  />
+            </run>
+        </groups>
+        <classes>
+            <class name="madalinapopescu.tests.tema7.CuraHealthcareLoghinTests"/>
+        </classes>
+    </test>
```
Comment (95teo) [inline src/test/resources/madalinapopescu/groups.xml:15]: You're missing closing suite tag - </suite> at the end of the file. Please add </suite> after </test> line. Thanks!

File: src/test/java/madalinapopescu/tests/tema7/CuraHealthcareMakeAppointmentTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+public class CuraHealthcareMakeAppointmentTest extends BaseTest{
+    @DataProvider(name = "appointmentData")
+    public Object [][] appointmentInfo(){
+        return new Object[][]{
+                {"03.02.2023","Programarea mea","validCredentials"},
+                {"01.2023","Programarea mea","invalidCredentials"},
+        };
+    }
+    public void loghinToMakeAppointment(){
```
Comment (95teo) [inline src/test/java/madalinapopescu/tests/tema7/CuraHealthcareMakeAppointmentTest.java:21]: Please leave one blank line between methods. Thanks!

File: src/test/java/madalinapopescu/tests/tema7/CuraHealthcareLoghinTests.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import org.testng.annotations.Test;
+public class CuraHealthcareLoghinTests extends BaseTest{
+    @DataProvider(name = "loginDataProvider")
+    public Object[][] loginDataProvider() {
+        return new Object[][]{
+                {"John Doe", "ThisIsNotAPassword", "validCredentials"},
+                {"John Do", "ThisIsNotAPassword", "invalidCredentials"}
+        };
+    }
+    @Test(dataProvider = "loginDataProvider", groups = "login")
```
Comment (95teo) [inline src/test/java/madalinapopescu/tests/tema7/CuraHealthcareLoghinTests.java:16]: Please leave one blank line between methods. Thanks!

File: src/test/java/madalinapopescu/tests/tema7/BaseTest.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import org.testng.annotations.BeforeMethod;
+public class BaseTest {
+    protected ChromeDriver driver;
+    @BeforeMethod(groups = "redirect")
+    public void initChromeBrowser() {
+        driver = BrowserManager.createChromeDriver();
+        driver.get("https://katalon-demo-cura.herokuapp.com/");
+        System.out.println("Executed before method");
+    }
+    @AfterMethod(groups = "redirect")
```
Comment (95teo) [inline src/test/java/madalinapopescu/tests/tema7/BaseTest.java:15]: Please leave one blank line between methods. Thanks!

File: src/test/java/madalinapopescu/tests/tema7/CuraHealthcareLoghinTests.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
+package madalinapopescu.tests.tema7;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+public class CuraHealthcareLoghinTests extends BaseTest{
```
Comment (95teo) [inline src/test/java/madalinapopescu/tests/tema7/CuraHealthcareLoghinTests.java:8]: Can you please rename the class to CuraHealthcareLoginTests? Be careful to update testng .xml files with suite/groups too. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

**Comment by 95teo** [comment]
Comments are marked as resolved, but I see no commit with those fixes. Have you made the push?

**Comment by pmadalina88** [comment]
As far as I remember I pushed, I checked again when I changed the brach. I did it once again as a precaution

File: src/test/resources/madalinapopescu/groups.xml (hunk 1 above)
Comment (95teo) [reply src/test/resources/madalinapopescu/groups.xml:15]: After </test> please add </suite>, thanks!

File: src/test/java/madalinapopescu/tests/tema7/loghinTestsCuraHealthcare.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
+package madalinapopescu.tests.tema7;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+public class loghinTestsCuraHealthcare extends BaseTest{
```
Comment (95teo) [inline src/test/java/madalinapopescu/tests/tema7/loghinTestsCuraHealthcare.java:8]: Please rename it to LoginTestsCuraHealthcare. Name of the classes start with capital letter in Java.

File: src/test/java/madalinapopescu/tests/tema7/loghinTestsCuraHealthcare.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+                    "https://katalon-demo-cura.herokuapp.com/#appointment",
+                    "Couldn't navigate to appointment page!");
+        } else {
+            WebElement unsuccessfulLoginMessage = driver.findElement(By.className("text-danger"));
+            Assert.assertEquals(unsuccessfulLoginMessage.getText(),
+                    "Login failed! Please ensure the username and password are valid.",
+                    "Shouldn't navigate to appointment page!");
+        }
+    }
+    @Test(groups = "redirect")
```
Comment (95teo) [inline src/test/java/madalinapopescu/tests/tema7/loghinTestsCuraHealthcare.java:43]: Please leave one blank line between methods. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

File: src/test/java/madalinapopescu/tests/tema7/loghinTestsCuraHealthcare.java (hunk 6 above)
Comment (pmadalina88) [reply src/test/java/madalinapopescu/tests/tema7/loghinTestsCuraHealthcare.java:8]: Am modificat numele

File: src/test/java/madalinapopescu/tests/tema7/loghinTestsCuraHealthcare.java (hunk 7 above)
Comment (pmadalina88) [reply src/test/java/madalinapopescu/tests/tema7/loghinTestsCuraHealthcare.java:43]: Am lasat spatiu liber....

File: src/test/resources/madalinapopescu/groups.xml (hunk 1 above)
Comment (pmadalina88) [reply src/test/resources/madalinapopescu/groups.xml:15]: <!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd" >

<suite name="SuitaTestNG" verbose="1" >

    <test name="CuraHealthcareLoghinTests">
        <classes>
            <class name="madalinapopescu.tests.tema7.LoghinTestsCuraHealthcare"/>
            <class name="madalinapopescu.tests.tema7.CuraHealthcareMakeAppointmentTest"/>
        </classes>
    </test>
</suite>

Comment (pmadalina88) [reply src/test/resources/madalinapopescu/groups.xml:15]: <suite name="SuitaTestNG" verbose="1" >

    <test name="CuraHealthcareLoghinTests">
        <classes>
            <class name="madalinapopescu.tests.tema7.LoghinTestsCuraHealthcare"/>
            <class name="madalinapopescu.tests.tema7.CuraHealthcareMakeAppointmentTest"/>
        </classes>
    </test>
</suite>

Comment (pmadalina88) [reply src/test/resources/madalinapopescu/groups.xml:15]: La mine apare si suita inchisa

Comment (pmadalina88) [reply src/test/resources/madalinapopescu/groups.xml:16]: Gata, acum am pus-o si in grupuri.... m-am prins mai greu :D

Comment (95teo) [reply src/test/resources/madalinapopescu/groups.xml:16]: great! thanks!

**Pull request merged**